packages/roosterjs-editor-core/lib/coreApi/createPasteFragment.ts

```diff
@@ -67,9 +67,12 @@ export const createPasteFragment: CreatePasteFragment = (
         }, event.htmlAttributes);
 
         clipboardData.htmlFirstLevelChildTags = [];
-        doc?.body.childNodes.forEach(node =>
-            clipboardData.htmlFirstLevelChildTags.push(getTagOfNode(node))
-        );
+        doc?.body.childNodes.forEach(node => {
+            const nodeTag = getTagOfNode(node);
```

Comment:
what if there is a valid text node as well?

suggest do doc?.body.normalize() then empty text node will be removed.

Reply:
It seems like normalize did not remove nodes with \r\n, I added a replace statement to remove this characters.
Also added a test case when the clipboard data have textNodes with Value and a IMG Tag